users/models.py

```diff
@@ -1,3 +1,246 @@
+    """Custom user manager for the User model."""
+
+    def create_user(self, email, password=None, **extra_fields):
+        """
+        Create and save a new user with the given email and password.
+
+        Args:
+            email (str): User's email address.
+            password (str, optional): User's password.
+            **extra_fields: Additional fields for the user.
+
+        Returns:
+            User: Newly created user instance.
+
+        Raises:
+            ValueError: If email is not provided or password is not set.
+        """
+        if not email:
+            raise ValueError('The Email field must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        if password:
+            user.set_password(password)
+            validate_password(password, user)
+        else:
+            raise ValueError('Password must be provided')
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password=None):
+        """
+        Create and save a new superuser with the given email and password.
+
+        Args:
+            email (str): Superuser's email address.
+            password (str, optional): Superuser's password.
+
+        Returns:
+            User: Newly created superuser instance.
+        """
+        user = self.create_user(
+            email,
+            password=password,
+            first_name='Admin',
+            last_name='User',
+            user_phone='',
+            is_staff=True,
+            is_superuser=True,
+            is_active=True,
+            role=User.ADMIN
+        )
+        return user
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    """
+    Custom user model extending Django's AbstractBaseUser and PermissionsMixin.
+
+    This model includes fields for user authentication, personal information,
+    and role management. It supports multiple roles for users, soft delete
+    functionality, and custom methods for role manipulation.
+    """
+
+    USER = 1
+    INVESTOR = 2
+    STARTUP = 4
+    ADMIN = 8
+
+    ROLE_CHOICES = [
+        (USER, 'User'),
+        (INVESTOR, 'Investor'),
+        (STARTUP, 'Startup'),
+        (ADMIN, 'Admin'),
+    ]
+
+    id = models.AutoField(primary_key=True)
+    email = models.EmailField(unique=True, max_length=255, db_index=True)
+    password = models.CharField(max_length=255)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    user_phone = models.CharField(
+        max_length=20,
+        validators=[RegexValidator(
+            regex=r'^\+?1?\d{9,15}$',
+            message="Phone number must be entered in the format: '+999999999'. Up to 15 digits allowed."
+        )]
+    )
+    role = models.PositiveSmallIntegerField(choices=ROLE_CHOICES, default=USER, db_index=True)
+    profile_picture = models.ImageField(upload_to='profile_pics/', blank=True, null=True)
```

Comment:
The field allows `null=True` and `blank=True`, but the file system might not handle null values well. It’s better to use only `blank=True` and store empty strings or default values instead.